code/modules/jobs/job_types/security.dm

```diff
@@ -124,7 +124,7 @@ Detective
 
 	outfit = /datum/outfit/job/detective
 
-	access = list(access_sec_doors, access_forensics_lockers, access_morgue, access_maint_tunnels, access_court, access_brig, access_weapons)
+	access = list(access_sec_doors, access_security, access_forensics_lockers, access_morgue, access_maint_tunnels, access_court, access_brig, access_weapons)
```

Comment:
I expect the player to know when it's acceptable to use their gun. They're not without their radio and police baton, either. If you think putting someone in a cell for 1 minute is better than taking them into interrogation and/or just giving them a stern warning, then I don't know what to tell you.